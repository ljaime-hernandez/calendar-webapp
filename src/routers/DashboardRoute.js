import {Routes} from 'react-router-dom';
import { CalendarScreen } from '../components/calendar/CalendarScreen';

export const DashboardRoutes = () => {
    return (
        <>
        
        <CalendarScreen/>

        <h1>Dashboard</h1>
            <Routes>
                
            </Routes>
        </>
    )
}