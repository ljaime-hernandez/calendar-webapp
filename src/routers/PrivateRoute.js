import { Navigate } from 'react-router-dom';

export const PrivateRoute = ({children, uid}) => {

    // The PrivateRoute component will check of uid existence, if the value
    // is not null then the user will be sent to its 'children' component which 
    // is the CalendarScreen page, if not the user will be redirected to the 
    // LoginScreen page
    return !!uid
        ?  children
        :  <Navigate to="/login"/>
}