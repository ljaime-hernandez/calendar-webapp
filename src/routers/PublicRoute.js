import { Navigate } from 'react-router-dom';

export const PublicRoute = ({ children, uid }) => {

    // the PublicRoute component will check of uid existence, if the value
    // is not null then the user will be redirected to the home page, if not it will
    // return to the PublicRoute 'children' component which is the LoginScreen page
    return !!uid
        ?  <Navigate to="/"/>
        :  children
}