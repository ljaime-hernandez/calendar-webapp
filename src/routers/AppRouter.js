import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { PublicRoute } from './PublicRoute'
import { PrivateRoute } from './PrivateRoute'
import { startChecking } from '../actions/auth'
import { LoginScreen } from '../components/auth/LoginScreen'
import { CalendarScreen } from '../components/calendar/CalendarScreen'

export const AppRouter = () => {

  const dispatch = useDispatch();
  // as soon as the webapp is launched, the useSelector will try to retrieve the checking status of
  // the page and the users uid, if the uid exists then the user will be redirected to the PrivateRoute
  // by default which is the '/' or home of the page, if the uid does not exists then the user will be
  // sent to the /login portion of the page to either introduce its credentials or create a user
  const { checking, uid } = useSelector( state => state.auth );

  useEffect(() => {
    
    // the useEffect will automatically dispatch a request to our calendar backend to check for 
    // 'communication' with the server, if it does not reach it, then the page will display an
    // h1 tag with a 'Wait' message which will only change after the server is back online
    dispatch(startChecking());

  }, [dispatch]);

  if(checking){
    return (<h1>Wait...</h1>);
  }
  
  return (
    <BrowserRouter>
            <div>
              <Routes>

                <Route path="/login" element={
                  <PublicRoute uid={uid}>
                    <LoginScreen/>
                  </PublicRoute>
                }/>

                <Route path="/*" element={
                  <PrivateRoute uid={uid}>
                    <CalendarScreen/>
                  </PrivateRoute>
                }/>

              </Routes>
            </div>
        </BrowserRouter>
  )
}
