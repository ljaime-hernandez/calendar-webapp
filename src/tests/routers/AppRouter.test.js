/* to run this test:
1. run the 'npm install' command from the calendar-webapp folder 
2. run the 'npm install --save-dev enzyme' command (if you have not done so)
3. run the 'npm install --save-dev enzyme-to-json' command (if you have not done so)
4. run the 'npm install --save-dev @wojtekmaj/enzyme-adapter-react-17 --legacy-peer-deps' command (if you are using React 17 as i do)
5. run the 'npm install --save-dev @testing-library/react-hooks' command (if you have not done so)
6. run the 'npm install redux-mock-store --save-dev' command (if you have not done so)
7. make sure the setupTests.js file include the enzyme, enzyme-to-json and the react adapter libraries
8. make sure you are running the calendar-backend as some tests depends on it
9. run the command 'npm run test'
10. to have a clearer view of this single js test file, press p. then type the file name 'AppRouter.test.js'
*/

import '@testing-library/jest-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { AppRouter } from '../../routers/AppRouter';

// for our mockStore we need to configure its middleware as a normal store so thunk must be used
// as in a normal store
const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

// the initial state of the auth reducer will be empty but will be modified according to the
// required test
const initState = {
    auth: {
        checking: true
    }
};

// the initial state of the store will be then filled with the prefixed initial state
let store = mockStore(initState);
store.dispatch = jest.fn();

describe('Tests on AppRouter component', () => {
    
    test('should render h1 default component', () => { 
        
        const initState = {
            auth: {
                checking: true
            }
        };

        let store = mockStore(initState);

        const wrapper = mount(
            <Provider
                store={store}
            >
                <AppRouter/>
            </Provider>
        )
        // the snapshot should display an h1 tag with a 'Wait' message as the default
        // initialState for this test is not communicating with the calendar backend
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').exists()).toBe(true);
    });

    test('should render public routes', () => { 
        
        const initState = {
            auth: {
                checking: false,
                uid: null
            }
        };
        
        let store = mockStore(initState);

        const wrapper = mount(
            <Provider
                store={store}
            >
                <AppRouter/>
            </Provider>
        );

        // the snapshot should display a LoginScreen component with a container class as the default
        // initialState for this test is emulating the communication with the calendar backend, we can
        // also test with any of the classes or elements present on the LoginScreen component
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.container').exists()).toBe(true);
    });

    test('should render private routes', () => { 
        
        const initState = {
            auth: {
                checking: false,
                uid: '123',
                name: 'Miguel'
            },
            calendar: {
                events: [],
                activeEvent: null
            },
            ui: {
                modalOpen: false
            }
        };
        
        let store = mockStore(initState);

        const wrapper = mount(
            <Provider
                store={store}
            >
                <AppRouter/>
            </Provider>
        )

        // the snapshot should display a CalendarScreen component with a navbar class as the default
        // initialState for this test is emulating the communication with the calendar backend on an
        // authorized user, we can also test with any of the classes or elements present on the 
        // NavBar component
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBe(true);
    });
});