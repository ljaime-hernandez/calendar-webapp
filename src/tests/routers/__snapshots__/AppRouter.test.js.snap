// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AppRouter component should render h1 default component 1`] = `
<h1>
  Wait...
</h1>
`;

exports[`Tests on AppRouter component should render private routes 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        Miguel
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key="#features"
            disabled={false}
            href="#features"
            onClick={[Function]}
          >
            Features
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="#pricing"
            disabled={false}
            href="#pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </div>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rr-ui-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
          >
            More deets
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="2"
            disabled={false}
            href="/"
            onClick={[Function]}
          >
            Dank memes
          </a>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            <i
              className="fa-solid fa-arrow-right-from-bracket"
            />
            Logout
          </button>
        </div>
      </div>
    </nav>
    <div
      className="rbc-calendar"
      style={
        Object {
          "height": 700,
        }
      }
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Today
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          junio 2022
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Month
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Week
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Day
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        aria-label="Month View"
        className="rbc-month-view"
        role="table"
      >
        <div
          className="rbc-row rbc-month-header"
          role="row"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              lun.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              mar.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              mié.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              jue.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              vie.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              sáb.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span
              aria-sort="none"
              role="columnheader"
            >
              dom.
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_0"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  30
                </button>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_1"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  31
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  01
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  02
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  03
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  04
                </button>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_6"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  05
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  06
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  07
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  08
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  09
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  10
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  11
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  12
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  13
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  14
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  15
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  16
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  17
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  18
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  19
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  20
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  21
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  22
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  23
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  24
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  25
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  26
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
          role="rowgroup"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            className="rbc-row-content"
            role="row"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  27
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  28
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  29
                </button>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  30
                </button>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_4"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  01
                </button>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_5"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  02
                </button>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_6"
                role="cell"
              >
                <button
                  className="rbc-button-link"
                  onClick={[Function]}
                  role="cell"
                  type="button"
                >
                  03
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </button>
  </div>
</div>
`;

exports[`Tests on AppRouter component should render public routes 1`] = `
<div>
  <div
    className="container mt-5 pt-5"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Access
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginEmail"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value="email@email.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="122345"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerName"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value="Miguel"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerEmail"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value="email@email.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="122345"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerConfirmPassword"
              onChange={[Function]}
              placeholder="Confirm your password"
              type="password"
              value="122345"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Create Account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
